addOns/ascanrulesBeta/src/main/java/org/zaproxy/zap/extension/ascanrulesBeta/XXEPlugin.java

```diff
@@ -293,19 +299,13 @@ public void scan() {
 
             try {
                 String requestBody = msg.getRequestBody().toString();
-                Matcher matcher = tagPattern.matcher(requestBody);
+                Matcher matcher;
                 StringBuilder sb = new StringBuilder(ATTACK_HEADER);
```

Comment:
This could be moved to the new method as well, so that the creation of the (whole) payload is kept in the same place.